pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -109,9 +119,34 @@ public static List<NativeQuery> parseJdbcSql(String query, boolean standardConfo
           i = Parser.parseBlockComment(aChars, i);
           break;
 
-        case '$': // possibly dollar quote start
-          i = Parser.parseDollarQuotes(aChars, i);
-          break;
+        case '$': { // possibly dollar quote start or a native placeholder
+          int end = Parser.parseDollarQuotes(aChars, i);
+          if (end == i && (
+              processParameters
+              && currentCommandType.supportsParameters()
+              && placeholderSetting.isAcceptedBySetting(ParameterContext.BindStyle.NATIVE))) {
+            // look for a native placeholder instead.
+
+            nativeSql.append(aChars, fragmentStart, i - fragmentStart);
+            end = Parser.parseNativePlaceholder(aChars, i);
+            if (end != i) {
+              fragmentStart = i;
+
+              // Keep the dollar in the captured name.
+              final String bindName = query.substring(fragmentStart, end + 1);
+              final int bindIndex = paramCtx.addNamedParameter(nativeSql.length(), ParameterContext.BindStyle.NATIVE, bindName);
+              NativeQuery.appendBindName(nativeSql, bindIndex);
+
+              i = end;
+              fragmentStart = i + 1;
+            } else {
+              fragmentStart = i;
+            }
+          } else {
+            i = end;
+          }
```

Comment:
In most cases, placing the shortest `if` branch first makes the code easier to follow.